plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -2370,9 +2380,12 @@ private void updateTableStatistics(
         }
 
         analyzedColumns.ifPresent(analyzeColumns -> {
-            if (!mergedColumnStatistics.keySet().equals(analyzeColumns)) {
+            Set<String> analyzePhysicalColumns = analyzeColumns.stream()
+                    .map(columnName -> physicalColumnNameMapping.map(mapping -> mapping.get(columnName)).orElse(columnName))
+                    .collect(toImmutableSet());
```

Comment:
do `physicalColumnNameMapping.map(mapping ->` first; if there is no mapping, nothing to do

also, when `mapping` is provided, is should map all columns, so remove defaulting to `columnName` when map entry not found